docs/decisions/005-ballista-extensions.md

```diff
@@ -22,115 +23,156 @@ Related decisions:
+* mTLS requires certificate management infrastructure (CA, cert rotation, distribution).
+  * **Mitigation**: Provide documentation and tooling for certificate generation. Consider integration with cert-manager in Kubernetes environments.
+
+#### Executor-to-Scheduler Signaling (rejected)
+
+* Original design had executors signal readiness to scheduler via oneshot channel.
+  * **Problem**: Limits HA scheduler designs—each scheduler would need to reliably receive the signal.
+  * **Resolution**: Use scheduler-initiated health checks instead.
 
 
 ## Consequences
 
-- Spice will maintain a fork of Apache Ballista at `github.com/spiceai/datafusion-ballista` with the extensions.
-- The Ballista fork will need to be kept in sync with upstream Apache Ballista releases, particularly during DataFusion upgrades.
-- New RPC endpoints will be added to Ballista's protobuf definitions:
-  - `GetCatalog` - Returns catalog, schema, and table metadata with Arrow schemas.
-  - `GetFunctions` - Returns UDF signatures, return types, and documentation.
-- `SessionContextExt` in the Ballista client will expose new builder methods:
-  - `remote_with_catalog()` - Connects to scheduler and populates catalog.
-  - `populate_functions_from_scheduler()` - Fetches and registers remote UDFs.
-  - `with_ballista_grpc_metadata()` - Configures metadata headers for authentication.
-  - `with_ballista_override_create_grpc_client_endpoint()` - Customizes gRPC endpoints for TLS.
-- `SchedulerConfig` and `ExecutorConfig` will expose endpoint customization hooks for TLS configuration.
-- `RemoteTableProvider` and `RemoteScalarUDF` will be introduced as stub implementations that error if execution is attempted locally (they must be executed on the cluster).
-- `BallistaLogicalExtensionCodec` will handle serialization/deserialization of remote stubs, resolving them to concrete implementations when queries reach the scheduler.
-- The executor's `poll_loop` function will accept an optional `oneshot::Sender` to signal readiness after the first successful poll.
-- Integration testing will need to cover TLS-enabled cluster scenarios and remote metadata synchronization.
-- Documentation will be needed for configuring TLS and API key authentication in distributed Spice deployments.
-- The development team will need processes for managing the fork, tracking upstream changes, and potentially upstreaming features.
+### Fork Management
+
+* Spice will maintain a fork of Apache Ballista at `github.com/spiceai/datafusion-ballista` with the extensions.
+* The Ballista fork will need to be kept in sync with upstream Apache Ballista releases, particularly during DataFusion upgrades.
+* The development team will need processes for managing the fork, tracking upstream changes, and potentially upstreaming features.
+* TLS/mTLS extensions are likely candidates for upstreaming to Apache Ballista.
```

Comment:
We'll also need changes to the executor registration, which right now for ballista is hard-coded to call back to a single scheduler